generators/app/templates/typescript/auth/src/index.ts

```diff
@@ -26,29 +29,59 @@ export default class AuthCustomPlugin implements IPluginAuth<CustomConfig> {
    * @param password provided password
    * @param cb callback function
    */
-  public authenticate(user: string, password: string, cb: Callback): void {
-    // here your code
+  public authenticate(user: string, password: string, cb: AuthCallback): void {
+    if (this.foo) {
+      cb(null, ['group-foo', 'group-bar']);
+    } else {
+      cb('error, try again', false);
+    }
   }
 
   /**
-   * check grants for such user.
+   * Triggered on each access request
+   * @param user
+   * @param pkg
+   * @param cb
    */
   public allow_access(
     user: RemoteUser,
     pkg: PackageAccess,
-    cb: Callback
+    cb: AuthCallback
   ): void {
-    // in case of restrict the access
+    /**
+     * This code is just an example for demostration purpose
+     */
+    if (user.name === this.foo && pkg.access.includes[user.name]) {
+      this.logger.debug({name: user.name}, 'your package has been granted for @{name}');
+      cb(null, ['group-foo', 'another-group-foo'])
+    } else {
+      this.logger.error({name: user.name}, '@{name} is not allowed to access this package');
+      cb('error, try again', false);
+    }
```

Comment:
Will it make sense to add the example code inside the comment?

```js
/**
 * This code is just an example for demostration purpose
 * if (user.name === this.foo && pkg.access.includes[user.name]) {
 *   this.logger.debug(
 *     { name: user.name },
 *     "your package has been granted for @{name}"
 *   );
 * cb(null, ["group-foo", "another-group-foo"]);
 *   } else {
 *    this.logger.error(
 *      { name: user.name },
 *      "@{name} is not allowed to access this package"
 *  );
 *  cb("error, try again", false);
 * }
 */
```

Reply:
Yes but, might be in the header, why I wan to achieve is provide the basic compile code so when run yarn build the project compiles. I can move the full example to the header as comment and leave below the most basic response, it is ok ?

Reply:
I do see the point of kickstarting with a compile friendly code. I am fine with either way.
both seem fine to me 👍

Reply:
better now?

Reply:
Yeah 👍